### PR TITLE
imprv: replace float based CSS to Grid based CSS of third party plugins

### PR DESCRIPTION
**Pull request by sagarb90**
### Description
**WooCommerce**
-  Replace float CSS to Grid layout CSS for Shop page, Single product page, Category and reviews. 
- Replace float based CSS to flex CSS of Shop page pagination. 
- We have make separate CSS files for this change and manage backward compatibility as well.

**EDD**
- Replace float CSS to Grid layout CSS for Shop/products page, Products Downloads (archive) page products listing and pagination markup  on all devices.
- We have make separate CSS files for this change and manage backward compatibility as well.

**LearnDash**
- We are conditionally loading Legacy related CSS from dynamic CSS.

**LifterLMS**
 -  Removed extra selectors of course listing CSS.
- Manage backward compatibility So for new user lifeterlms-flex.css file will be loading.
-  Single lesson static CSS move to dynamic and load conditionally on is_lesson() condition .
- My courses grid layout page -  removed extra selectors.

### Types of changes
 - Improvements
 
### How has this been tested?
Verify third party plugins on all devices i.e  WooCommerce, EDD and Learndahs for all page design layouts

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

**Comment by premanshup** [comment]
@sagarb90 The bottom spacing on page Shop of WooCommerce is not appearing properly.

Steps to reproduce:

1. Set the following settings in Customizer - [https://d.pr/i/qSICvU](https://d.pr/i/qSICvU) for both the previous version and this branch.

2. Check the shop page for both:

Previous version - [https://d.pr/i/hkWNHT](https://d.pr/i/hkWNHT)

Current branch - [https://d.pr/i/gMzSc8](https://d.pr/i/gMzSc8)

**Comment by premanshup** [comment]
@sagarb90  Please check the spacing of `Next Course` link in the current branch and older versions for `Lifter LMS`.

Previous versions: [https://d.pr/i/CyfYKY](https://d.pr/i/CyfYKY)
Current branch: [https://d.pr/i/nYth0M](https://d.pr/i/nYth0M)

Please check the alignment as well.

Also, check this for both the logged-in and logged-out users.

**Comment by premanshup** [comment]
@sagarb90 Lifter LMS - Sign up page border issue:

Previous versions: [https://d.pr/i/Fqhf52](https://d.pr/i/Fqhf52)
Current version: [https://d.pr/i/VQY0wj](https://d.pr/i/VQY0wj)

Please check!

File: sass/site/compatibility/lifterlms-flex.scss [hunk 1]
```diff
@@ -0,0 +1,1148 @@
+/**
+ * lifterlms.scss
+ * Governs the general look and feel of WooCommerce sections of stores using themes that do not
```
Comment (premanshup) [inline sass/site/compatibility/lifterlms-flex.scss:3]: @sagarb90 Please check the comment.

File: sass/site/compatibility/lifterlms-flex.scss [hunk 2]
```diff
@@ -0,0 +1,1148 @@
...
+	}	
+}
+
+.ast-woocommerce-cart-menu {
+	.main-header-bar{
+		.main-header-log-out {
+		    padding-left: 10px;
+		}
+	}
+}
```
Comment (premanshup) [inline sass/site/compatibility/lifterlms-flex.scss:1148]: @sagarb90 Why is WooCommerce CSS added in the lifter LMS file?

File: inc/compatibility/edd/class-astra-edd.php [hunk 3]
```diff
@@ -444,10 +444,6 @@ public function astra_get_edd_cart() {
 
 			$view_shopping_cart = apply_filters( 'astra_edd_view_shopping_cart_title', __( 'View your shopping cart', 'astra' ) );
 			$edd_cart_link      = apply_filters( 'astra_edd_cart_link', edd_get_checkout_uri() );
-
-			if ( is_customize_preview() ) {
-				$edd_cart_link = '#';
-			}
```
Comment (premanshup) [inline inc/compatibility/edd/class-astra-edd.php:450]: @sagarb90 I think this is required, please confirm this with @RajkiranBagal

File: inc/compatibility/edd/class-astra-edd.php [hunk 4]
```diff
@@ -484,7 +480,7 @@ public function astra_get_edd_cart() {
 		 * @since 1.5.5
 		 */
 		public function add_styles( $assets ) {
-			$assets['css']['astra-edd'] = 'compatibility/edd';
+			$assets['css']['astra-edd'] = 'compatibility/edd-grid';
```
Comment (premanshup) [inline inc/compatibility/edd/class-astra-edd.php:483]: @sagarb90 Have we not added backward compatibility for `compatibility/edd`?

File: sass/site/compatibility/learndash.scss [hunk 5]
```diff
@@ -5,1082 +5,709 @@
...
+ 
+ // LearnDash Quiz dragable elements not working scrolling the page on mobile devices.
+ body.single-sfwd-quiz {
+	 overflow-x: unset;
  }
-
-// LearnDash Quiz dragable elements not working scrolling the page on mobile devices.
-body.single-sfwd-quiz {
-	overflow-x: unset;
-}
```
Comment (premanshup) [inline sass/site/compatibility/learndash.scss:1086]: @sagarb90 Just confirming if we have checked this with LearnDash old version Legacy as well?

**Review by premanshup — CHANGES_REQUESTED** [review]

File: sass/site/compatibility/lifterlms-flex.scss (hunk 2 above)
Comment (sagarb90) [reply sass/site/compatibility/lifterlms-flex.scss:1148]: @premanshup This is not newly added by me It's already their for lifter LMS dependent woo cart related CSS, please refer this - https://share.getcloudapp.com/yAu65xDg

**Comment by sagarb90** [comment]
> @sagarb90 The bottom spacing on page Shop of WooCommerce is not appearing properly.
> 
> Steps to reproduce:
> 
> 1. Set the following settings in Customizer - https://d.pr/i/qSICvU for both the previous version and this branch.
> 2. Check the shop page for both:
> 
> Previous version - https://d.pr/i/hkWNHT
> 
> Current branch - https://d.pr/i/gMzSc8
> 
> Below CSS is missing in this branch:
> 
> `.woocommerce ul.products li.product, .woocommerce-page ul.products li.product { float: left; margin: 0 20px 2.992em 0; padding: 0; position: relative; width: 22.05%; margin-left: 0; }`

@premanshup Yes I have resolved this. and also confirm with Previous version.

File: sass/site/compatibility/learndash.scss (hunk 5 above)
Comment (sagarb90) [reply sass/site/compatibility/learndash.scss]: @premanshup This code was not removing It just change line number so it is showing change/code removed.

File: inc/compatibility/edd/class-astra-edd.php (hunk 4 above)
Comment (sagarb90) [reply inc/compatibility/edd/class-astra-edd.php]: @premanshup Yes I had missed it , have added backward compatibility.

File: inc/compatibility/edd/class-astra-edd.php (hunk 3 above)
Comment (sagarb90) [reply inc/compatibility/edd/class-astra-edd.php]: Yes their was  code conflicts so unexpectedly it was removed now updated with latest code.

File: sass/site/compatibility/lifterlms-flex.scss (hunk 1 above)
Comment (sagarb90) [reply sass/site/compatibility/lifterlms-flex.scss]: @premanshup Yes corrected the same.

**Comment by sagarb90** [comment]
> @sagarb90 Lifter LMS - Sign up page border issue:
> 
> Previous versions: https://d.pr/i/Fqhf52
> Current version: https://d.pr/i/VQY0wj
> 
> Please check!

@premanshup Yes resolved border issue It was because of we have changed float based to flex based CSS.

**Comment by sagarb90** [comment]
> @sagarb90 Please check the spacing of `Next Course` link in the current branch and older versions for `Lifter LMS`.
> 
> Previous versions: https://d.pr/i/CyfYKY
> Current branch: https://d.pr/i/nYth0M
> 
> Please check the alignment as well.
> 
> Also, check this for both the logged-in and logged-out users.

@premanshup Yes It is because of a. post-navigation  CSS missing  in earlier/previous 3.0.0 version while optimize ast-pagination CSS, So I have added back, now in current branch It 's correct and in Previous version it was missing .https://share.getcloudapp.com/RBu4Qdrm

File: sass/site/compatibility/lifterlms-flex.scss (hunk 2 above)
Comment (premanshup) [reply sass/site/compatibility/lifterlms-flex.scss:1149]: @sagarb90 I see! Thank you for sharing the screenshot.

File: inc/compatibility/edd/class-astra-edd.php (hunk 3 above)
Comment (premanshup) [reply inc/compatibility/edd/class-astra-edd.php]: @sagarb90 Thank you!

File: inc/compatibility/edd/class-astra-edd.php (hunk 4 above)
Comment (premanshup) [reply inc/compatibility/edd/class-astra-edd.php]: @sagarb90 Thank you!

File: sass/site/compatibility/learndash.scss (hunk 5 above)
Comment (premanshup) [reply sass/site/compatibility/learndash.scss]: @sagarb90 Got it!

**Comment by premanshup** [comment]
> > @sagarb90 Please check the spacing of `Next Course` link in the current branch and older versions for `Lifter LMS`.
> > Previous versions: https://d.pr/i/CyfYKY
> > Current branch: https://d.pr/i/nYth0M
> > Please check the alignment as well.
> > Also, check this for both the logged-in and logged-out users.
> 
> @premanshup Yes It is because of a. post-navigation CSS missing in earlier/previous 3.0.0 version while optimize ast-pagination CSS, So I have added back, now in current branch It 's correct and in Previous version it was missing .https://share.getcloudapp.com/RBu4Qdrm

So is this issue since v3.0.0? When we optimized the pagination CSS? Can you please confirm?

**Review by premanshup — APPROVED** [review]

**Review by vrundakansara — APPROVED** [review]

**Review by rohitp-bsf — APPROVED** [review]

**Pull request merged**